lib/foreman_tasks/tasks/generate_task_actions.rake

```diff
@@ -0,0 +1,41 @@
+#
+# generate_task_actions.rake is a data migration task to properly fill the missing
+# values into the foreman_tasks_tasks table.
+#
+# Run "foreman-rake foreman_tasks:generate_task_actions" to generate missing values.
+
+namespace :foreman_tasks do
+  desc 'Generate missing values for action column in foreman_tasks_tasks table.'
+
+  BATCH_SIZE = 100
+
+  task :generate_task_actions => :environment do
+    class ProgressReporter
+      def initialize(count, message = nil)
+        @count = count
+        @processed = 0
+        puts message % { :count => count } unless message.nil?
+      end
+
+      def progress(count)
+        @processed += count
+      end
+
+      def report
+        puts _('Processed %{processed}/%{count} tasks') % { :processed => @processed, :count => @count }
+      end
+    end
+
+    scope = ::ForemanTasks::Task.where(:action => nil)
```

Comment:
What about ordering by `started_at desc`? This way, when somebody runs this and cuts it in the middle (because she doesn't want to wait until the script to finish), she still can search in the most recent/relevant tasks.